[bug] build requirement fails to build with new CMakeDeps

### Describe the bug

## Summary

`conan install` fails to build requirement when `--build-require` is specified and `tools.cmake.cmakedeps:new=will_break_next` is used.

## Expected Behaviour

Using the following conanfile
```.py
from conan import ConanFile

required_conan_version = ">=2.0.9"

class MyToolchain(ConanFile):
    name = "my-toolchain"
    settings = "os", "arch", "compiler", "build_type"

    def requirements(self):
        self.requires("libxml2/2.15.0")
```

and the following profile:
```
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu20
compiler.libcxx=libstdc++11
compiler.version=15
os=Linux

[conf]
tools.cmake.cmakedeps:new=will_break_next
tools.cmake.cmaketoolchain:generator=Ninja
```

The command line
```
conan install . --build-require --profile:build=build_profile --profile:host=build_profile --build='libxml2/*'
```
should successfully install the required packages as a build requirement.

## Actual Behaviour

libxml2 fails to build with 
```
CMake Error at CMakeLists.txt:346 (target_link_libraries):
  Target "LibXml2" links to:

    Iconv::Iconv
```

## How to reproduce it

Using the profile above, run

```
conan install . --build-require --profile:build=build_profile --profile:host=build_profile --build='libxml2/*'
```

## Context

The intent of the recipe is to build a toolchain from sources. The toolchain has various dependencies, such as libxml2, which are provided by conan. The toolchain will be consumed downstream as described in [Toolchain Packages](https://docs.conan.io/2/examples/cross_build/toolchain_packages.html)

CMakeConfigDeps is used as the CMake build is complex and it requires fewer patches to the source than CMakeDeps.

## System Environment

OS: Arch Linux x86_64 
Kernel: 6.17.9-arch1-1
Conan version 2.24.0
Python 3.13.3

--- a/conan/tools/cmake/cmakedeps2/target_configuration.py
+++ b/conan/tools/cmake/cmakedeps2/target_configuration.py
@@ -121,7 +121,7 @@ class TargetConfigurationTemplate2:
 
         libs = {}
         # The BUILD context does not generate libraries targets atm
-        if self._conanfile.context == CONTEXT_HOST:
+        if not self._require.build:
             libs = self._get_libs(cpp_info, pkg_name, pkg_folder, pkg_folder_var)
             self._add_root_lib_target(libs, pkg_name, cpp_info)
         exes = self._get_exes(cpp_info, pkg_name, pkg_folder, pkg_folder_var)

--- a/test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py
+++ b/test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py
@@ -1002,8 +1002,10 @@ class TestToolRequires:
         c.run("new header_lib -d name=hello -d version=1.0 -o=hello")
         c.run("create hello -tf=")
         c.run("new cmake_lib -d name=bye -d version=1.0 -d requires=hello/1.0 -o=bye")
-        c.run(f"install bye --build-require -c:a tools.cmake.cmakedeps:new={new_value}")
-        cmake = c.load("bye/build/generators/hello-TargetsBuild-release.cmake")
+        # Ninja for same layout in all platforms
+        c.run(f"install bye --build-require -c:a tools.cmake.cmakedeps:new={new_value} "
+              f"-c:a tools.cmake.cmaketoolchain:generator=Ninja")
+        cmake = c.load("bye/build/Release/generators/hello-TargetsBuild-release.cmake")
         assert "add_library(hello::hello INTERFACE IMPORTED)" in cmake
 
 

--- a/test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py
+++ b/test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py
@@ -997,6 +997,15 @@ class TestToolRequires:
         assert 'set(tool_INCLUDE_DIR' not in tool_config
         assert 'set(tool_LIBRARIES' not in tool_config
 
+    def test_libs_build_context(self):
+        c = TestClient()
+        c.run("new header_lib -d name=hello -d version=1.0 -o=hello")
+        c.run("create hello -tf=")
+        c.run("new cmake_lib -d name=bye -d version=1.0 -d requires=hello/1.0 -o=bye")
+        c.run(f"install bye --build-require -c:a tools.cmake.cmakedeps:new={new_value}")
+        cmake = c.load("bye/build/generators/hello-TargetsBuild-release.cmake")
+        assert "add_library(hello::hello INTERFACE IMPORTED)" in cmake
+
 
 @pytest.mark.tool("cmake")
 @pytest.mark.parametrize("tool_requires", [True, False])

